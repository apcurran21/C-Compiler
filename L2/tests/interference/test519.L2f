(@calculate
	1
	%n <- rdi
	%output <- 1
	%index <- 0
	:call_label_ciao__ciao_4
	%check <- %index = %n
	cjump %check = 1 :call_label_ciao__ciao_5
	%index += 1
	%output *= %index
	goto :call_label_ciao__ciao_4
	:call_label_ciao__ciao_5
	rax <- %output
	return
)
