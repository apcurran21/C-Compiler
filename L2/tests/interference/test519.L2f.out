%check %index %n %output r12 r13 r14 r15 rbp rbx
%index %check %n %output r12 r13 r14 r15 rbp rbx
%n %check %index %output r12 r13 r14 r15 rbp rbx
%output %check %index %n r12 r13 r14 r15 rbp rbx
r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r11 r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r12 %check %index %n %output r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r13 %check %index %n %output r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r14 %check %index %n %output r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r15 %check %index %n %output r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx rsi
r8 r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi
r9 r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi
rax r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi
rbp %check %index %n %output r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx rsi
rbx %check %index %n %output r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx rsi
rcx r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi
rdi r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi
rdx r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi
rsi r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx
